src/hotspot/share/cds/cds_globals.hpp

```diff
@@ -71,6 +71,12 @@
   product(bool, AllowArchivingWithJavaAgent, false, DIAGNOSTIC,             \
           "Allow Java agent to be run with CDS dumping")                    \
                                                                             \
+  develop(ccstr, ArchiveHeapTestClass, NULL,                                \
+          "For JVM internal testing only. Not available in "                \
+          "product builds. If specified, the static field named "           \
+          "\"archivedObjects\" in this class is stored into the CDS "       \
+          "archive heap")                                                   \
```

Comment:
The “this class” wasn’t clear to me until I read the test case. I’d suggest something like the following:

```
“The "\"archivedObjects\" static field of the specified”
“class is stored in the CDS archive heap."
```

Reply:
Fixed as you suggested.